### PR TITLE
[newman] Adding missing fields to NewmanRunFailure.

### PR DESCRIPTION
**Pull request by FormerPhysicist**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: 

There are 3 fields missing from the NewmanRunFailure type: source, parent, cursor.  See link below.  These fields are present from newman 3.1.1 to the present version. Not being familiar with the newman or postman code base, I don't know the shapes of these fields, but I think they should at least be included with a type of 'any'

[location in code where they add failures](https://github.com/postmanlabs/newman/blob/b2208685bde2152546975a1e07ab21d9726eadb3/lib/run/summary.js#L319)

### CODE REVIEW

File: types/newman/index.d.ts
Comment (Graham42) [inline types/newman/index.d.ts]: My best estimate for this
```suggestion
    source: NewmanRunExecutionItem | undefined;
```

File: types/newman/index.d.ts
Comment (Graham42) [inline types/newman/index.d.ts]: A guess based on https://github.com/postmanlabs/newman/blob/d984e5d042f131bc43b4f359044f8b772e5b803d/test/unit/run-summary.test.js#L73
and other usage
https://github.com/postmanlabs/newman/blob/e08983dad7798531a4f42e3747b3a6d101ac949e/lib/run/summary.js#L238-L241
```suggestion
    cursor: { ref: string } | {};
```

File: types/newman/index.d.ts
Comment (Graham42) [inline types/newman/index.d.ts]: :heavy_check_mark:  `parent` is often referred to as any in the [postman collection types](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/postman-collection/index.d.ts) so this should be fine

**Review by Graham42 — CHANGES_REQUESTED** [review]
See inline

File: types/newman/index.d.ts
Comment (FormerPhysicist) [reply types/newman/index.d.ts]: @Graham42 Should it be `source?: NewmanRunExecutionItem`

Comment (FormerPhysicist) [reply types/newman/index.d.ts]: Not arguing btw, just want to know what the best practice/style is with typescript.

**Review by Graham42 — APPROVED** [review]
Thanks for the submission!

**Comment by FormerPhysicist** [comment]
@Graham42 Can I ask when this PR will get merged?  Are you the right person to ping?

**Comment by Graham42** [comment]
Sorry I can't merge it, the maintainers of the repo need to do that.

**Pull request merged**
